store/src/java/com/zimbra/cs/service/admin/CheckBlobConsistency.java

```diff
@@ -65,63 +67,82 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
 
         // Check blobs and assemble response.
         Element response = zsc.createElement(AdminConstants.CHECK_BLOB_CONSISTENCY_RESPONSE);
-
-        StoreManager sm = StoreManager.getInstance();
-        if (sm instanceof ExternalStoreManager) {
-
-            for (int mboxId : mailboxIds) {
-                ExternalBlobConsistencyChecker checker = new ExternalBlobConsistencyChecker();
-                BlobConsistencyChecker.Results results = checker.check(null, mboxId, checkSize, reportUsedBlobs);
-                if (results.hasInconsistency() || reportUsedBlobs) { //or checking used blobs
-                    Element mboxEl = response.addElement(AdminConstants.E_MAILBOX).addAttribute(AdminConstants.A_ID, mboxId);
-                    results.toElement(mboxEl);
+        List<Element> volumeElementList = request.listElements(AdminConstants.E_VOLUME);
+        List<Short> volumeIds = new ArrayList<Short>();
+        if (!volumeElementList.isEmpty()) {
```

Comment:
can we have null and empty check here

Reply:
As per implementation of `request.listElements(AdminConstants.E_VOLUME)` returned list never be null.